homeassistant/components/workday/binary_sensor.py

```diff
@@ -138,8 +138,10 @@ def setup_platform(
         _LOGGER.debug("No holidays to remove or invalid holidays")
 
     _LOGGER.debug("Found the following holidays for your configuration:")
-    for remove_holiday, name in sorted(obj_holidays.items()):
-        _LOGGER.debug("%s %s", remove_holiday, name)
+    for holiday_date, name in sorted(obj_holidays.items()):
+        # Make explicit str variable to avoid "Incompatible types in assignment"
+        _holiday_string = holiday_date.strftime("%Y-%m-%d")
```

Comment:
To avoid mypy getting confused as `holidays` now explicitly defines `HolidayBase` as `dict[date, str]` previously we were just using the default string representation of a `datetime.date`.